test/builtin/test_datentime.py

```diff
@@ -10,29 +10,22 @@
 import pytest
 
 
-@pytest.mark.skipif(
-    sys.platform in ("emscripten",),
-    reason="TimeConstrained is based in Threads, which are not supported in Piodide",
-)
 def test_timeremaining():
     str_expr = "TimeConstrained[1+2; TimeRemaining[], 0.9]"
     result = evaluate(str_expr)
     assert result is None or 0 < result.to_python() < 9
 
 
-@pytest.mark.skipif(
-    sys.platform in ("emscripten",),
-    reason="TimeConstrained is based in Threads, which are not supported in Piodide",
-)
 def test_timeconstrained1():
     #
     str_expr1 = "a=1.; TimeConstrained[Do[Pause[.01];a=a+1,{1000}],.1]"
     result = evaluate(str_expr1)
     str_expected = "$Aborted"
     expected = evaluate(str_expected)
     assert result == expected
+    assert evaluate("a").to_python() <= 11
```

Comment:
This kind of coding smells really bad.  It assumes something about the speed of computation.

Reply:
Indeed, it is fragile. In any case, what we actually need to test is if after `TimeConstrained` returns a value, the inner expression continues the evaluation.

Reply:
On the other hand, if `Pause` could ensure some degree of accurateness, and TimeConstrained does its job, I would expect that the number of increments would be bounded by some reasonable number.

Reply:
So let's spend the time to come up with a better test please.

Reply:
OK, but right now I would like to figure out why this test fails randomly in macos and not in the other platforms. Once I narrow the issue, maybe a better test come up. Of course, ideas for a better test are welcome.

Reply:
> OK, but right now I would like to figure out why this test fails randomly in macos and not in the other platforms.

A good test might help elucidate this.  

> Once I narrow the issue, maybe a better test come up. Of course, ideas for a better test are welcome.

Give me some time to think of something.